src/parsing.jl

```diff
@@ -351,6 +386,19 @@ function pqparse(::Type{ZonedDateTime}, ptr::Ptr{UInt8})
     return ZonedDateTime(dt, tz"UTC"; from_utc=true)
 end
 
+function pqparse(::Type{UTCDateTime}, ptr::Ptr{UInt8})
+    value = ntoh(unsafe_load(Ptr{Int64}(ptr)))
+    if value == typemax(Int64)
+        depwarn_timetype_inf()
+        return UTCDateTime(typemax(DateTime))
+    elseif value == typemin(Int64)
+        depwarn_timetype_inf()
+        return UTCDateTime(typemin(DateTime))
+    end
+    dt = POSTGRES_EPOCH_DATETIME + Microsecond(value)
+    return UTCDateTime(dt)
+end
```

Comment:
This can be:

```julia
function pqparse(::Type{UTCDateTime}, ptr::Ptr{UInt8})
    return UTCDateTime(pqparse(DateTime, ptr))
end
```

(seems like you can apply that `value` optimization to the `DateTime` method directly below; not sure why that wasn't there before given that it's present in the `Date` method)